src/datreant/core/state.py

```diff
@@ -331,6 +338,13 @@ def _serialize(self, state, handle):
 
 
 class JSONFile(FileSerial):
+    def __init__(self, filename, init_state, **kwargs):
+
+        super(JSONFile, self).__init__(filename, **kwargs)
+
+        # set _init_state function that given
+        self._init_state = init_state
```

Comment:
the object `self._init_state` is nowhere really called.

Reply:
``self._init_state`` gets called by ``Tags`` and ``Categories`` classes, which use ``JSONFile`` to talk to their state files. Basically, it's defined at runtime on creation of the ``JSONFile`` object.

I'll admit it's a confusing way to go about it. I'm open to alternatives.

Reply:
yeah, but where is it called? It should also be commented what the intent of this. Otherwise, I'm gonna ask again in the future.

Reply:
I added a brief docstring to each with this information. Is there more needed, though?

Reply:
Nope clearer now. A comment here may be nice. But I don't remember exactly why I didn't understand this. Reading the code now it's clear to me.